Readability.js

```diff
@@ -197,20 +197,29 @@ Readability.prototype = {
       return pathBase + uri;
     }
 
-    function convertRelativeURIs(tagName, propName) {
-      var elems = articleContent.getElementsByTagName(tagName);
-      this._forEachNode(elems, function(elem) {
-        var relativeURI = elem.getAttribute(propName);
-        if (relativeURI != null)
-          elem.setAttribute(propName, toAbsoluteURI(relativeURI));
-      });
-    }
-
-     // Fix links.
-    convertRelativeURIs.call(this, "a", "href");
+    var links = articleContent.getElementsByTagName("a");
+    this._forEachNode(links, function(link) {
+      var href = link.getAttribute("href");
+      if (href) {
+        // Replace links with javascript: URIs with text content, since
+        // they won't work after scripts have been removed from the page.
+        if (href.indexOf("javascript:") === 0) {
+          var text = new Text();
+          text.textContent = link.textContent;
+          link.parentNode.replaceChild(text, link);
```

Comment:
I updated this to be a bit less aggressive and only replace links with javascript: URLs with their text content. However, I was running into a jsdom error with replaceChild when testing locally:

  1) Test pages clean-links jsdom "before all" hook:
     TypeError: Object [object Object] has no method 'dispatchEvent'
      at core.Node.insertBefore (/Users/leibovic/code/readability/node_modules/jsdom/lib/jsdom/level2/events.js:332:14)
      at core.Node.replaceChild (/Users/leibovic/code/readability/node_modules/jsdom/lib/jsdom/level1/core.js:707:10)
      at Object.<anonymous> (/Users/leibovic/code/readability/Readability.js:209:27)
      at NodeList.forEach (native)
      at Object.Readability._forEachNode (/Users/leibovic/code/readability/Readability.js:149:36)
      at Object.Readability._fixRelativeUris (/Users/leibovic/code/readability/Readability.js:201:10)
      at Object.Readability._postProcessContent (/Users/leibovic/code/readability/Readability.js:134:10)
      at Object.Readability.parse (/Users/leibovic/code/readability/Readability.js:1721:10)
      at /Users/leibovic/code/readability/test/test-readability.js:110:34
      at Context.<anonymous> (/Users/leibovic/code/readability/test/test-readability.js:17:16)
      at callFn (/Users/leibovic/code/readability/node_modules/mocha/lib/runnable.js:266:21)
      at Hook.Runnable.run (/Users/leibovic/code/readability/node_modules/mocha/lib/runnable.js:259:7)
      at next (/Users/leibovic/code/readability/node_modules/mocha/lib/runner.js:268:10)
      at Object._onImmediate (/Users/leibovic/code/readability/node_modules/mocha/lib/runner.js:289:5)
      at processImmediate [as _immediateCallback] (timers.js:345:15)

I'm not sure what's going on here. @n1k0 @gijsk ideas?